src/main/java/tdl/record/sourcecode/content/CopyFromDirectorySourceCodeProvider.java

```diff
@@ -1,29 +1,90 @@
 package tdl.record.sourcecode.content;
 
+import java.io.File;
 import org.apache.commons.io.FileUtils;
 
 import java.io.IOException;
 import java.nio.file.Path;
 import org.apache.commons.io.filefilter.IOFileFilter;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.lib.ObjectId;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.revwalk.DepthWalk.RevWalk;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.revwalk.RevTree;
+import org.eclipse.jgit.treewalk.TreeWalk;
 import tdl.record.sourcecode.snapshot.helpers.ExcludeGitDirectoryFileFilter;
+import tdl.record.sourcecode.snapshot.helpers.GitHelper;
 
 public class CopyFromDirectorySourceCodeProvider implements SourceCodeProvider {
 
     private final Path sourceFolderPath;
 
     private final IOFileFilter filter;
 
+    private Git git;
+
     public CopyFromDirectorySourceCodeProvider(Path sourceFolderPath) {
         this.sourceFolderPath = sourceFolderPath;
         filter = new ExcludeGitDirectoryFileFilter(sourceFolderPath);
+        initGitIfAvailable();
+    }
+
+    private void initGitIfAvailable() {
+        if (!GitHelper.isGitDirectory(sourceFolderPath)) {
```

Comment:
I think you were right in thinking that we should have a separate sourceCodeProvider. This provider should become a selector (wrapper) and select either normal copy or git tree walk

Reply:
I introduced new class for handling the Git. This current class will use that class if the file handled is a git directory.